File: src/main/scala/robosky/ether/world/feature/minidungeons/MinidungeonFeatureConfig.scala
package robosky.ether.world.feature.minidungeons

import java.util.Optional

import net.minecraft.util.Identifier

// Does not extend FeatureConfig so we can get it into the StructureStart because this shit is dumb
// Uses java Optional because we need to get a value out from a java class
case class MinidungeonFeatureConfig(name: String, template: Identifier, loot: Optional[Identifier])

File: src/main/java/robosky/uplands/world/feature/minidungeons/MinidungeonFeatureConfig.java
package robosky.uplands.world.feature.minidungeons;

import com.mojang.serialization.Codec;
import com.mojang.serialization.codecs.RecordCodecBuilder;

import net.minecraft.util.Identifier;
import net.minecraft.world.gen.feature.FeatureConfig;

public final class MinidungeonFeatureConfig implements FeatureConfig {

    public static final Codec<MinidungeonFeatureConfig> CODEC = RecordCodecBuilder.create(inst -> inst.group(
        Identifier.CODEC.fieldOf("template").forGetter(c -> c.template),
        Identifier.CODEC.fieldOf("loot").forGetter(c -> c.loot)
    ).apply(inst, MinidungeonFeatureConfig::new));

    public final Identifier template;
    public final Identifier loot;

    public MinidungeonFeatureConfig(Identifier template, Identifier loot) {
        this.template = template;
        this.loot = loot;
    }
}
